templates/npmrc/.npmrc

```diff
@@ -0,0 +1 @@
+legacy-peer-deps=true
```

Comment:
Why do we need this flag? It will ignore peer dependency conflicts during installation, which might cause headaches during runtime, could we document what exactly are we trying to solve with this flag?

Reply:
Yes! Agreed, this probably is not a great fix. Everything works well with `yarn` and `bun`, only `npm` causes problems. I added the entire 2nd commit to address this [test suite](https://github.com/thoughtbot/belt/actions/runs/18972633560/job/54183917672) failure. Can you explore if there's a better way to solve that test failure?